src/main/scala/EnsimePlugin.scala

```diff
@@ -453,11 +453,14 @@ object EnsimePlugin extends AutoPlugin {
 
     val IvyConfig = "([A-Za-z]+)->([A-Za-z]+)".r
 
-    val deps = project.dependencies.map { d =>
+    def depsFor(config: Configuration): Seq[EnsimeProjectId] = project.dependencies.map { d =>
       val name = d.project.project
-      val config = d.configuration.collect { case IvyConfig(_, to) => to }.getOrElse("compile")
-      (name, config)
-    }.map(t => EnsimeProjectId(t._1, t._2))
+      val con = d.configuration match {
+        case Some(a) => a.split(";").collect { case IvyConfig(from, to) => (from, to) }
+        case None => Array(("compile", "compile"))
+      }
+      con.filter(_._1 == config.name).map(p => EnsimeProjectId(name, p._2))
+    }.flatten
```

Comment:
I think this could be better, I'd appreciate any suggestions.

Reply:
I would go for (maybe less readable): 
```
def depsFor(config: Configuration) = project.dependencies.flatMap { d =>
  lazy val name = d.project.project
  d.configuration.map { a =>
    a.split(";").collect { 
      case IvyConfig(from, to) if from == config.name => EnsimeProjectId(name, to)
    }
  }.getOrElse(Array(EnsimeProjectId("compile", "compile")))
}
```
`flatMap` and absence of `filter`/`map` is my main point.

Reply:
Thanks for the tip. I've used flatMap to get rid of the map, flatten and filter. 

I still prefer to use pattern matching on the inner map to keep it readable.